### PR TITLE
Provide alternative implementation for registerBlock without reflection

### PR DESCRIPTION
**Pull request by diesieben07**
Like #509 but targeted at 1.8 and without all the reflection.

### CODE REVIEW

File: src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java [hunk 1]
```diff
@@ -210,12 +205,39 @@ public static Block registerBlock(Block block, Class<? extends ItemBlock> itemcl
...
+     */
+    public static Block registerBlock(Block block, ItemBlock item, String name)
+    {
+        if (Loader.instance().isInState(LoaderState.CONSTRUCTING))
+        {
+            FMLLog.warning("The mod %s is attempting to register a block whilst it it being constructed. This is bad modding practice - please use a proper mod lifecycle event.", Loader.instance().activeModContainer());
+        }
+        try
+        {
+            assert block != null : "registerBlock: block cannot be null";
```
Comment (jk-5) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Maybe replace this with a normal null check, throwing a nice exception if the block is null?

File: src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java [hunk 2]
```diff
@@ -191,13 +191,8 @@ public static Block registerBlock(Block block, Class<? extends ItemBlock> itemcl
      */
     public static Block registerBlock(Block block, Class<? extends ItemBlock> itemclass, String name, Object... itemCtorArgs)
     {
-        if (Loader.instance().isInState(LoaderState.CONSTRUCTING))
-        {
-            FMLLog.warning("The mod %s is attempting to register a block whilst it it being constructed. This is bad modding practice - please use a proper mod lifecycle event.", Loader.instance().activeModContainer());
-        }
         try
         {
-            assert block != null : "registerBlock: block cannot be null";
```
Comment (jk-5) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Same here?

Comment (diesieben07) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: I just left in what FML already had.

Comment (jk-5) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: If a null block is registered it's going to crash anyway, so we might as well be descriptive

Comment (diesieben07) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: If you so wish... I'll update it :P

Comment (diesieben07) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Happy now? :)

Comment (jk-5) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java]: Lol. I'm happy now :+1:

**Comment by LexManos** [comment]
The problem with this is that it needs duplicate protection.
I can just see it now, some modder is gonna register all of their blocks with the same Item instance...

**Comment by diesieben07** [comment]
Ok, I'll put in a check for that in a bit.

File: src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java [hunk 3]
```diff
@@ -191,13 +192,8 @@ public static Block registerBlock(Block block, Class<? extends ItemBlock> itemcl
      */
     public static Block registerBlock(Block block, Class<? extends ItemBlock> itemclass, String name, Object... itemCtorArgs)
```
Comment (SoniEx2) [inline src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java:193]: Shouldn't this be deprecated?

**Pull request closed without merging**